app/src/main/java/com/android/sample/MainActivity.kt

```diff
@@ -85,11 +85,13 @@ fun ActivitiesApp(uid: String, startDestination: String) {
   val navController = rememberNavController()
   val navigationActions = NavigationActions(navController)
 
-  val listActivitiesViewModel: ListActivitiesViewModel =
-      viewModel(factory = ListActivitiesViewModel.Factory)
+  // val listActivitiesViewModel: ListActivitiesViewModel =
+  //  viewModel(factory = ListActivitiesViewModel.Factory)
   val profileViewModel: ProfileViewModel = hiltViewModel()
   // need to add factory for SignInViewModel
   val authViewModel: SignInViewModel = hiltViewModel()
+  val locationViewModel: LocationViewModel = hiltViewModel()
+  val listActivitiesViewModel: ListActivitiesViewModel = hiltViewModel()
```

Comment:
Good job adding the viewmodel for location and using hilt to instantiate the viewmodel listactivities